### PR TITLE
Constraint fixes

### PR DESCRIPTION
**Pull request by mikegioia**
This update handles the constraint corner-cases that @dimko prepared in the test file. The additional logic from PR #253 handles monthly calendars but didn't address the oddities for weekly calendars. For example, some weekly calendars would render without any days/weeks because the constraints conflicted with the interval.

This PR addresses all cases for monthly and weekly calendars and all tests should now pass. I'd invite a second pair of eyes and I know the constraint logic in lines 239 to 320 is somewhat confusing. It should be commented enough to give some clarity though :stuck_out_tongue:

### CODE REVIEW

**Comment by mikegioia** [comment]
@dimko I finished reviewing your PR and added to it to handle the problems with weekly calendars we identified. If it's possible, could you pull this down and run through the tests to see if everything is working okay? I ran through all of them and it looks good to go. Thank you again for this update and the test cases that you added, it really helped identify the deficiencies in the constraint logic.

File: src/clndr.js [hunk 1]
```diff
@@ -205,10 +211,11 @@
                 if (this.options.lengthOfTime.startDate) {
                     this.intervalStart =
                         moment(this.options.lengthOfTime.startDate)
-                            .startOf('day');
+                            .startOf('week');
```
Comment (mikegioia) [inline src/clndr.js]: I changed to this to `week` to enforce any errors arising from users trying to set a weekly calendar to the middle of the week. That causes breakage in the calendar so this should correct any of those errors. cc @kylestetz

Comment (kylestetz) [inline src/clndr.js]: This breaks a use case that I think is valid and useful... Using `lengthOfTime.days` you should be able to create a calendar that spans an arbitrary number of days (2, 3, 5, etc.). This hardcodes the assumption that the user wants a calendar specified in _weeks_, and now it will not honor the `lengthOfTime.startDate` option (instead going to the beginning of that week).

I tested this by making a 3 day calendar and setting the `lengthOfTime.startDate` to the middle of the week. It starts on the previous Sunday instead.

Does that make sense? Am I missing anything?

Comment (mikegioia) [inline src/clndr.js]: Yea @kylestetz that's correct. And ok, I'll revert it. I didn't know that was something this calendar supported :stuck_out_tongue: I just noticed it when I typed a random date into the test file and it broke what should have been a weekly calendar.

Comment (kylestetz) [inline src/clndr.js]: I get that a 3 day calendar has questionable usefulness, but this problem would extend to a calendar that only shows 1 day at a time. That feels like a more useful example. If we can support it we should try!

File: tests/test.html [hunk 2]
```diff
@@ -75,19 +75,37 @@
 
         <p>
             <strong>clndr.constraints</strong> Test start and end constraints.
-            (4th of this month to the 12th of next month). Logs in the console.
+            (the 4th of this month to the 12th of next month). Logs in the console.
         </p>
         <div id="constraints" class="cal1"></div>
 
         <p>
+            <strong>clndr.prevNextMonthConstriants</strong> Test start and end constraints.
```
Comment (kylestetz) [inline tests/test.html]: Sorry to nitpick :grin: some of these are misspelled `constriant` and I don't want anyone to get confused if they copy & paste them!

Comment (mikegioia) [inline tests/test.html]: @kylestetz I just pushed up those typo fixes. I'm going to test this branch once more but I think this is good to go and resolves a few bugs too.

**Comment by kylestetz** [comment]
LGTM :+1:
